identity-service/src/routes/trackListens.js

```diff
@@ -249,15 +249,26 @@ module.exports = function (app) {
         if (!hourlyResponseData.hasOwnProperty(hourSuffix)) {
           hourlyResponseData[hourSuffix] = {
             submission: Number(await redis.get(trackingRedisKeys.submission)),
-            success: Number(await redis.get(trackingRedisKeys.success))
+            success: Number(await redis.get(trackingRedisKeys.success)),
+            time: hourSuffix
           }
           submission += hourlyResponseData[hourSuffix].submission
           success += hourlyResponseData[hourSuffix].success
         }
       }
     }
 
-    return successResponse({ success, submission, hourlyResponseData })
+    // Sort response in descending time order
+    let keys = Object.keys(hourlyResponseData)
+    let sortedKeys = []
+    let data = []
+    if (keys) {
+      sortedKeys = keys.sort((a, b) => (new Date(b) - new Date(a)))
+      for (var sortedKey of sortedKeys) {
+        data.push(hourlyResponseData[sortedKey])
+      }
+    }
```

Comment:
This is fine, but you could also do this somewhat nicely with some mapping

```
const data =
  Object.keys(hourlyResponseData)
    .sort((a, b) => (new Date(b) - new Date(a)))
    .map(key => hourlyResponseData[key])
```

Reply:
love this